core/indexer/processTransactions.go

```diff
@@ -140,6 +136,22 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 	return append(convertMapTxsToSlice(transactions), rewardsTxs...), alteredAddresses
 }
 
+func getGasUsedFromReceipt(rec *receipt.Receipt, tx *Transaction) uint64 {
+	if rec.Data != nil && string(rec.Data) == transactionProcess.RefundGasMessage {
+		gasUsed := big.NewInt(0).SetUint64(tx.GasPrice)
+		gasUsed.Mul(gasUsed, big.NewInt(0).SetUint64(tx.GasLimit))
+		gasUsed.Sub(gasUsed, rec.Value)
+		gasUsed.Div(gasUsed, big.NewInt(0).SetUint64(tx.GasPrice))
+
+		return gasUsed.Uint64()
+	}
+
+	gasUsed := big.NewInt(0)
```

Comment:
Maybe add a comment, that in this case `the receipt contains the cost (the fee)`.

Reply:
done.